src/ExpressionEvaluator/Core/Source/ResultProvider/Formatter.cs

```diff
@@ -71,23 +73,23 @@ string IDkmClrFullNameProvider.GetClrTypeName(DkmInspectionContext inspectionCon
             return sawInvalidIdentifier ? null : name;
         }
 
-        string IDkmClrFullNameProvider.GetClrArrayIndexExpression(DkmInspectionContext inspectionContext, int[] indices)
+        string IDkmClrFullNameProvider.GetClrArrayIndexExpression(DkmInspectionContext inspectionContext, string[] indices)
         {
             return GetArrayIndexExpression(indices);
         }
 
-        string IDkmClrFullNameProvider.GetClrCastExpression(DkmInspectionContext inspectionContext, string argument, DkmClrType type, DkmClrCustomTypeInfo customTypeInfo, bool parenthesizeArgument, bool parenthesizeEntireExpression)
+        string IDkmClrFullNameProvider.GetClrCastExpression(DkmInspectionContext inspectionContext, string argument, DkmClrType type, DkmClrCustomTypeInfo customTypeInfo, DkmClrCastExpressionOptions castExpressionOptions)
         {
             bool sawInvalidIdentifier;
             var name = GetTypeName(new TypeAndCustomInfo(type, customTypeInfo), escapeKeywordIdentifiers: true, sawInvalidIdentifier: out sawInvalidIdentifier);
             if (sawInvalidIdentifier)
```

Comment:
`GetTypeName(type, ..., out sawInvalidIdentifier)` will set `sawInvalidIdentifier` if the metadata name of the type contains characters that are not valid in a C# or VB identifier.